integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/config/IntegrasjonspunktHandlerExceptionResolver.java

```diff
@@ -0,0 +1,67 @@
+package no.difi.meldingsutveksling.config;
+
+import lombok.RequiredArgsConstructor;
+import no.difi.meldingsutveksling.exceptions.HttpStatusCodeException;
+import org.eclipse.jetty.io.EofException;
+import org.springframework.context.MessageSource;
+import org.springframework.context.NoSuchMessageException;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.mvc.support.DefaultHandlerExceptionResolver;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.ConstraintViolationException;
+import java.io.IOException;
+import java.util.Arrays;
+
+@Component
+@RequiredArgsConstructor
+public class IntegrasjonspunktHandlerExceptionResolver extends DefaultHandlerExceptionResolver {
+
+    private final MessageSource messageSource;
+
+    @Override
+    protected ModelAndView doResolveException(HttpServletRequest request, HttpServletResponse response, Object handler, Exception ex) {
+
+        try {
+            if (ex instanceof HttpStatusCodeException) {
+                return handleHttpStatusCodeException(
+                        (HttpStatusCodeException) ex, request, response);
+            } else if (ex instanceof ConstraintViolationException) {
+                return handleConstraintViolationException(
+                        (ConstraintViolationException) ex, response);
+            }
+        } catch (EofException e) {
+            if (logger.isWarnEnabled()) {
+                logger.warn("Handling of [" + ex.getClass().getName() + "] resulted in exception: " + e.getCause().getLocalizedMessage());
+            }
+        } catch (Exception handlerException) {
+            if (logger.isWarnEnabled()) {
+                logger.warn("Handling of [" + ex.getClass().getName() + "] resulted in exception", handlerException);
+            }
+        }
+
+        return super.doResolveException(request, response, handler, ex);
+    }
+
+    private ModelAndView handleConstraintViolationException(ConstraintViolationException ex, HttpServletResponse response) throws IOException {
+        response.sendError(HttpStatus.BAD_REQUEST.value(), ex.getLocalizedMessage());
+        return new ModelAndView();
+    }
+
+    private ModelAndView handleHttpStatusCodeException(HttpStatusCodeException ex,
+                                                       HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.sendError(ex.getStatusCode().value(), getMessage(ex, request));
+        return new ModelAndView();
+    }
+
+    private String getMessage(HttpStatusCodeException ex, HttpServletRequest request) {
+        try {
+            return messageSource.getMessage(ex.getMessage(), ex.getArgs(), request.getLocale());
+        } catch (NoSuchMessageException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)